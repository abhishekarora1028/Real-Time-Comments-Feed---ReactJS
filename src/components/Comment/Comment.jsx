import React from 'react';
import * as PropTypes from 'prop-types';
import { Media} from 'react-bootstrap';
import './Comment.scss';

const Comment = (props) => {
	const { comment } = props;
	return (
		<Media data-id={comment.id} className="commentbox mb-2">
              <div className="author-image">{comment.name.substring(0,1)}</div>
              <Media.Body>
                <p className="comment" data-testid="commentmessage">
                  {comment.message}
                </p>
                <p className="author">By {comment.name} at {comment.created}</p>
              </Media.Body>
    </Media>
	);
};

Comment.propTypes = {
	comment: PropTypes.object.isRequired,
}


export default Comment;