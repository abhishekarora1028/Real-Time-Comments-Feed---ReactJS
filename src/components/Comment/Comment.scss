.author-image {
    box-shadow: inset 0 0 0 2px var(--primary);
    text-transform: uppercase;
    font-weight: 500;
    width: 64px;
    height: 64px;
    object-fit: cover;
    display: flex;
    justify-content: center;
    border-radius: 50%;
    align-items: center;
    margin-right: 1rem;
}
.author {
    font-size: 12px;
    font-style: italic;
    color: var(--gray);
    text-align: right;
}
.commentbox {
    border: 1px solid var(--gray);
    padding: 10px 10px 0px 10px;
    border-radius: 10px;
}
.comment {
    text-align: left;
}