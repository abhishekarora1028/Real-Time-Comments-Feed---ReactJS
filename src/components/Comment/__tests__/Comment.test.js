import React from 'React';
import { render, cleanup } from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect";
import Comment from '../Comment';

afterEach(cleanup);

describe('Comment', () => {
    const defaultProps = {
        comment: {"name":"abhishek", "message":"this is test message", "created":"2021-08-16 08:41:39"}
    };

    it('Should render the comment box successfully', () =>{
        const { asFragment } = render(<Comment {...defaultProps} />);
        expect(asFragment(<Comment {...defaultProps}/>)).toMatchSnapshot();
    });

    it('Should have the comment message', () =>{
        const { getByTestId } = render(<Comment {...defaultProps} />);
        expect(getByTestId('commentmessage')).toHaveTextContent('this is test message');
    });
});