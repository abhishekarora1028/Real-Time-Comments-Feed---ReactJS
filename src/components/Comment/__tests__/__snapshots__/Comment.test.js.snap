// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Comment Should render the comment box successfully 1`] = `
<DocumentFragment>
  <div
    class="commentbox mb-2 media"
  >
    <div
      class="author-image"
    >
      a
    </div>
    <div
      class="media-body"
    >
      <p
        class="comment"
        data-testid="commentmessage"
      >
        this is test message
      </p>
      <p
        class="author"
      >
        By abhishek at 2021-08-16 08:41:39
      </p>
    </div>
  </div>
</DocumentFragment>
`;
